### PR TITLE
analog: add end-to-end test for quadrature demod block

### LINKED ISSUE #2020: Quadrature Demod: missing FM input end-to-end QA test
So, we need a unit test with FM modulated signals getting decoded correctly in a zero-noise scenario.

extend gr-analog/python/analog/qa_quadrature_demod.py:

1. add a new `def test_fm_end_to_end(self):`
2. in that, use the `analog.frequency_modulator_fc` block with, 
3. implement 
```text 
signal source (cos 1000 Hz) +-> vector_sink "A"
                            |
                            \-> analog.frequency_modulator_fc -> analog.quadrature_demod_cf -> vector_sink "B"
```
4. verify that, first samples aside, A.data() is equal to B.data() after running

### PR DESCRIPTION
**Pull request by ffu**
## Description
Add unit test for the quadrature demod block to verify that it can decode frequency modulated signals.

## Related Issue
Fixes #2020

## Which blocks/areas does this affect?
analog.quadrature_demod_cf

## Testing Done
Ran "make test" before and after changes to ensure all tests pass

## Checklist
- [X] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [X] I have squashed my commits to have one significant change per commit. 
- [X] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [X] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [X] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Review by marcusmueller — APPROVED** [review]
Seems a welcome addition to test coverage, so that we don't accidentally screw up any sensitivity calculation.

File: gr-analog/python/analog/qa_quadrature_demod.py [hunk 1]
```diff
@@ -48,6 +48,24 @@ def test_quad_demod_001(self):
...
         self.assertComplexTuplesAlmostEqual(expected_result, result_data, 5)
 
+    def test_fm_end_to_end_002(self):
+        f = 1000.0
+        fs = 8000.0
+        src = analog.sig_source_f(fs, analog.gr_waveform_t.GR_COS_WAVE, f, 1.0)
+        head = blocks.head(gr.sizeof_float, 100)
+        sensitivity = cmath.pi / 4
+        fm = analog.frequency_modulator_fc(sensitivity)
+        quad_demod = analog.quadrature_demod_cf(1.0/sensitivity)
```
Comment (marcusmueller) [inline gr-analog/python/analog/qa_quadrature_demod.py:58]: Our Python style checker [tells us](https://github.com/gnuradio/gnuradio/actions/runs/12603799768/job/35129742848?pr=7618#step:4:8)  that you forgot the spaces around the `/` operator. Please fix!

**Review by marcusmueller — CHANGES_REQUESTED** [review]

File: gr-analog/python/analog/qa_quadrature_demod.py (hunk 1 above)
Comment (ffu) [reply gr-analog/python/analog/qa_quadrature_demod.py]: No problem. I completed your change request.

**Comment by willcode** [comment]
Could someone check that the `expected_result` and `result_data` have meaningful looking data in them?

**Comment by ffu** [comment]
Here's a plot of result_data. As expected, it's a cosine wave, except the first value is 0.0 due to initial conditions of the quadrature demod block. The expected_result is similar.

result-data

**Comment by willcode** [comment]
Sorry, I meant to say the modulated signal. The test vector would be just what it says it is :smile: 

Anyway, I'll check it out myself, since I was just curious whether the modulated signal came out to pretty much a constant.
